lib/Dialect/TTNN/IR/TTNNOpsAttrs.cpp

```diff
@@ -962,6 +975,20 @@ ShardSpecAttr::getCoreRangeSet(mlir::MLIRContext *context,
   return CoreRangeSetAttr::get(context, coreRangeSet);
 }
 
+NDShardSpecAttr NDShardSpecAttr::get(TTNNNDLayoutAttr layout) {
+  auto shardGrid = layout.getGrid();
+  auto context = layout.getContext();
```

Comment:
:warning: **llvm\-qualified\-auto** :warning:
`` auto context `` can be declared as `` auto *context ``
```suggestion
  auto *context = layout.getContext();
```